File: java-framework/src/main/java/examples/jobsearch/SimpleHtmlWriter.java
package examples.jobsearch;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.PrintWriter;
import java.util.List;
import java.util.Properties;

public class SimpleHtmlWriter {

    public static String templateFileName = "examples/jobsearch/html_template.properties";
    public static Properties templates = new Properties();
    public static String indent = "  ";

    static {
        InputStream is = SimpleHtmlWriter.class.getClassLoader().getResourceAsStream(templateFileName);
        if (is == null) {
            throw new RuntimeException("Cannot find template resource file: " + templateFileName);
        }
        try {
            templates.load(is);
        } catch (IOException e) {
            e.printStackTrace();
            //throw new RuntimeException(e.getMessage());
        }
    }

    public PrintWriter out;

    public SimpleHtmlWriter(OutputStream out) {
        this.out = new PrintWriter(out);
    }

    public void close() {
        this.out.flush();
        this.out.close();
    }

    public SimpleHtmlWriter beginHtml() {
        out.println("<html>");
        String head = templates.getProperty("head", "<head/>");
        out.println(head);
        out.println("<body>");
        return this;
    }

    public SimpleHtmlWriter endHtml() {
        out.println("</body>");
        out.println("</html>");
        return this;
    }

    public SimpleHtmlWriter write(String str) {
        out.print(str);
        return this;
    }

    public SimpleHtmlWriter beginTable() {
        out.println("<table>");
        return this;
    }

    public SimpleHtmlWriter endTable() {
        out.println("</table>");
        return this;
    }

    public SimpleHtmlWriter writeTableHead(List<String> headingList) {
        out.print(indent);
        out.println("<thead><tr>");
        for(String heading : headingList) {
            out.print(indent);
            out.print(indent);
            out.print("<th>");
            out.print(heading);
            out.println("</th>");
        }
        out.print(indent);
        out.println("</tr></thead>");
        return this;
    }

    public SimpleHtmlWriter writeTableRow(List<String> cellDataList) {
        out.print(indent);
        out.println("<tr>");
        for(String cellData : cellDataList) {
            out.print(indent);
            out.print(indent);
            out.print("<td>");
            out.print(cellData);
            out.println("</td>");
        }
        out.print(indent);
        out.println("</tr>");
        return this;
    }

}

File: cs-examples/CaStateJobsSearch/SimpleHtmlWriter.cs
using System.IO;
using System.Collections.Generic;

namespace WebUI.Automation.Examples.CaStateJobsSearch
{

    public class SimpleHtmlWriter
    {

        public string HtmlIndent { get; set; } = "  ";
        public string HtmlDoctype { get; set; } = "<!DOCTYPE html>";
        public string HtmlHead { get; set; } = "<head/>";

        private TextWriter writer;

        public SimpleHtmlWriter(string outputFileName)
        {
            writer = new StreamWriter(outputFileName);
        }

        public virtual void Close()
        {
            writer.Flush();
            writer.Close();
        }

        public virtual SimpleHtmlWriter BeginHtml()
        {
            writer.WriteLine(HtmlDoctype);
            writer.WriteLine("<html>");
            writer.WriteLine(HtmlHead);
            writer.WriteLine("<body>");
            return this;
        }

        public virtual SimpleHtmlWriter EndHtml()
        {
            writer.WriteLine("</body>");
            writer.WriteLine("</html>");
            return this;
        }

        public virtual SimpleHtmlWriter Write(string str)
        {
            writer.Write(str);
            return this;
        }

        public virtual SimpleHtmlWriter WriteLine(string str)
        {
            writer.WriteLine(str);
            return this;
        }

        public virtual SimpleHtmlWriter BeginTable()
        {
            writer.WriteLine("<table>");
            return this;
        }

        public virtual SimpleHtmlWriter EndTable()
        {
            writer.WriteLine("</table>");
            return this;
        }

        public virtual SimpleHtmlWriter WriteTableHead(IList<string> headingList)
        {
            writer.Write(HtmlIndent);
            writer.WriteLine("<thead><tr>");
            foreach (string heading in headingList)
            {
                writer.Write(HtmlIndent);
                writer.Write(HtmlIndent);
                writer.Write("<th>");
                writer.Write(heading);
                writer.WriteLine("</th>");
            }
            writer.Write(HtmlIndent);
            writer.WriteLine("</tr></thead>");
            return this;
        }

        public virtual SimpleHtmlWriter WriteTableRow(IList<string> cellDataList)
        {
            writer.Write(HtmlIndent);
            writer.WriteLine("<tr>");
            foreach (string cellData in cellDataList)
            {
                writer.Write(HtmlIndent);
                writer.Write(HtmlIndent);
                writer.Write("<td>");
                writer.Write(cellData);
                writer.WriteLine("</td>");
            }
            writer.Write(HtmlIndent);
            writer.WriteLine("</tr>");
            return this;
        }

    }

}